File: src/main/java/es/uji/sape/model/Assignment.java
package es.uji.sape.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDate;

import static es.uji.sape.model.AssignmentState.UNDEFINED;

@Data
@NoArgsConstructor
public final class Assignment {

    private int projectOfferId;
    private @NotNull String studentCode = "";
    private @NotNull String tutorCode = "";
    private @NotNull LocalDate proposalDate = LocalDate.now();
    private @Nullable LocalDate acceptanceDate;
    private @Nullable LocalDate rejectionDate;
    private @Nullable LocalDate igluTransferDate;
    private @NotNull AssignmentState state = UNDEFINED;

}

File: src/main/kotlin/es/uji/sape/model/Assignment.kt
package es.uji.sape.model

import es.uji.sape.model.AssignmentState.UNDEFINED
import java.time.LocalDate

data class Assignment(
    val projectOfferId: Int = 0,
    val studentCode: String = "",
    val tutorCode: String = "",
    val proposalDate: LocalDate = LocalDate.now(),
    val acceptanceDate: LocalDate? = null,
    val rejectionDate: LocalDate? = null,
    val igluTransferDate: LocalDate? = null,
    val state: AssignmentState = UNDEFINED
)
